src/metabase/api/embed.clj

```diff
@@ -251,18 +251,50 @@
 
 ;;; -------------------------- Dashboard Fns used by both /api/embed and /api/preview_embed --------------------------
 
+(defn- remove-locked-and-disabled-param-values [dashboard embedding-params]
+  (let [params-to-remove (set (concat (for [[param status] embedding-params
+                                            :when          (not= status "enabled")]
+                                        param)
+                                      (for [{slug :slug} (:parameters dashboard)
+                                            :let         [param (keyword slug)]
+                                            :when        (not (contains? embedding-params param))]
+                                        param)))
+        params-ids-to-remove (set (for [parameter (:parameters dashboard)
+                                        :when     (contains? params-to-remove (keyword (:slug parameter)))]
+                                    (:id parameter)))
+        field-ids-to-remove (flatten
+                             (for [card (:dashcards dashboard)]
+                               (for [parameter-mappings (:parameter_mappings card)
+                                     :when              (contains? params-ids-to-remove (:parameter_id parameter-mappings))]
+                                 (-> (:target parameter-mappings)
+                                     (nth 1)
+                                     (nth 1)))))
```

Comment:
you can put multiple things inside of `for`, this should save the call to flatten, too:

```clojure
(for [x [[1 10] [2 20] [3 30]]]
  (for [y x]
    y))
;; => ((1 10) (2 20) (3 30))

(for [x [[1 10] [2 20] [3 30]]
      y x]
  y)
;; => (1 10 2 20 3 30)
```

I didn't run it, but I think this is still the same.

```suggestion
        field-ids-to-remove (for [card (:dashcards dashboard)
                                  parameter-mappings (:parameter_mappings card)
                                  :when (contains? params-ids-to-remove (:parameter_id parameter-mappings))]
                              (ffirst (:target parameter-mappings)))
```